src/Microsoft.AspNetCore.Server.Kestrel/Http/Connection.cs

```diff
@@ -32,23 +32,20 @@ public class Connection : ConnectionContext, IConnectionControl
         private readonly object _stateLock = new object();
         private ConnectionState _connectionState;
 
-        private IPEndPoint _remoteEndPoint;
-        private IPEndPoint _localEndPoint;
-
         public Connection(ListenerContext context, UvStreamHandle socket) : base(context)
         {
             _socket = socket;
             ConnectionControl = this;
 
-            _connectionId = Interlocked.Increment(ref _lastConnectionId);
+            ConnectionId = Interlocked.Increment(ref _lastConnectionId).ToString(CultureInfo.InvariantCulture);
```

Comment:
Keep in mind that this is once per connection, not per request. I think Frame's copy constructor would always resolve it anyways.